crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -517,9 +517,17 @@ where
 
     #[inline]
     fn initialize(&mut self, world: &mut World) {
-        self.world_id = Some(world.id());
-        self.system_meta.last_run = world.change_tick().relative_to(Tick::MAX);
-        self.param_state = Some(F::Param::init_state(world, &mut self.system_meta));
+        if let Some(id) = self.world_id {
+            assert_eq!(
+                id,
+                world.id(),
+                "System built with a different world than the one it was added to.",
+            );
+        } else {
+            self.world_id = Some(world.id());
+            self.system_meta.last_run = world.change_tick().relative_to(Tick::MAX);
```

Comment:
Do you need to update `last_run` like this in `SystemBuilder::build`, since it won't be updated here?